server/src/main/java/org/elasticsearch/search/internal/ExitableDirectoryReader.java

```diff
@@ -276,7 +276,7 @@ public void visit(int docID, byte[] packedValue) throws IOException {
 
         @Override
         public PointValues.Relation compare(byte[] minPackedValue, byte[] maxPackedValue) {
-            queryCancellation.checkCancelled();
+            checkAndThrowWithSampling();
```

Comment:
I don't think we should throttle this. We don't have enough evidence that it's slowing down the request so I'd keep it as is for now.